scaaml/capture/scope/lecroy/lecroy_waveform.py

```diff
@@ -521,3 +566,252 @@ def translate(attr_name: str, value: Union[int, str]) -> str:
+                                self.selected_lines_mask.append(False)
+                            case "1":
+                                self.selected_lines_mask.append(True)
+                            case _:
+                                raise ValueError(
+                                    "Unexpected character in StatusDescription"
+                                    f" {child.text}")
+                case "BusInfo":
+                    for bus_info in child:
+                        match bus_info.tag:
+                            case "Name" | "HasSymbolic":
+                                pass
+                            case "NumLevels":
+                                assert bus_info.text is not None
+                                assert int(
+                                    bus_info.text
+                                ) == 2, "Wrong number of binary levels"
+                            case "NumLines":
+                                assert bus_info.text is not None
+                                self.num_lines = int(bus_info.text)
+                case "LineInfo":
+                    line_infos = [
+                        DigitalChannelWaveform.Line.parse(line_element)
+                        for line_element in child
+                    ]
+                case "HorUnit":
+                    assert child.text is not None
+                    self.horizontal_unit = child.text
+                case "HorScale":
+                    for scale_info in child:
+                        match scale_info.tag:
+                            case "UniformInterval" | "ClockSynchronized":
+                                assert scale_info.text == "true"
+                            case "HorStart":
+                                pass
+                            case "HorPerStep":
+                                assert scale_info.text is not None
+                                self.horizontal_per_step = float(
+                                    scale_info.text)
+                            case "HorResolution":
+                                assert scale_info.text is not None
+                                self.horizontal_resolution = float(
+                                    scale_info.text)
+                            case _:
+                                raise ValueError(
+                                    f"Unknown HorScale {scale_info.tag}")
+                case "NumSamples":
+                    if child.text:
+                        self.num_samples = int(child.text)
+                    else:
+                        # child.text could be None, default
+                        self.num_samples = 0
+                case _:
+                    raise ValueError(f"Unknown ILecDigitalResult: {child.tag}")
+
+        # Check consistency of selected lines:
+        # - their number is the same
+        assert len(line_infos) == sum(
+            1 if x else 0 for x in self.selected_lines_mask
```

Comment:
`sum(int(x) for x in self.selected_lines_mask)` should do the same.

If you need to do manipulation/parsing/conversion of bit strings you can also have a look at the bitarray module.